core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -55,8 +62,11 @@ export class Question {
       inapplicableSkillMisconceptionIds);
   }
 
+  // Explicitly cast to string as getting id means that the question
+  // has been created in the backend. This is important as the temporary
+  // of the question id is null but getId should not return null in any case.
```

Comment:
Why shouldn't `getId` return `null`?

Reply:
There wasn't a case of getting Id of question while it was in temporary state, but adding null to return type also lead to other TS errors due to strict type checks. I now implemented this in another fashion. Please take a look. Thanks!